cb_maintenance/models/maintenance_request.py

```diff
@@ -21,18 +22,55 @@ class MaintenanceRequest(models.Model):
         default="unspecified",
         required=True,
     )
-    category_id = fields.Many2one(related=False)
-    solved = fields.Boolean(related="stage_id.done")
-    close_datetime = fields.Datetime("Close Date", readonly=True)
+    follower_id = fields.Many2one("res.users", readonly=True)
+    category_id = fields.Many2one(related=False, track_visibility="onchange")
+    close_datetime = fields.Datetime(
+        "Close Date", readonly=True, track_visibility="onchange"
+    )
     maintenance_team_id_member_ids = fields.Many2many(
         "res.users",
         relation="selectable_maintenance_members",
         compute="_compute_maintenance_team_id_member_ids",
     )
     technician_user_id = fields.Many2one(
-        domain="[('id', 'in', maintenance_team_id_member_ids)]", default=False
+        compute="_compute_technician_user_id", store=True
+    )
+    technician_id = fields.Many2one(
+        "res.partner",
+        domain="[('user_ids', 'in', maintenance_team_id_member_ids)]",
+        track_visibility="onchange",
+    )
+    manager_id = fields.Many2one(
+        "res.users",
+        string="Manager",
+        domain="[('id', 'in', maintenance_team_id_member_ids)]",
+        default=False,
+        track_visibility="onchange",
     )
-    manager_id = fields.Many2one("res.users", string="Manager")
+    maintenance_color = fields.Char(compute="_compute_color")
+    color = fields.Integer(compute="_compute_color", store=True)
+
+    @api.depends("maintenance_type")
+    def _compute_color(self):
+        for record in self:
+            # Hardcoded colors, green for preventive and red for corrective
+            c_preventive = 10
+            c_corrective = 1
+            if record.maintenance_type == "preventive":
```

Comment:
Maybe a hook that returns color and maintenance_color? In order to use properly with works...